orttraining/pytorch_frontend_examples/nvidia-bert/ort_supplement/ort_supplement.py

```diff
@@ -0,0 +1,128 @@
+import onnx
+from onnxruntime.capi.ort_trainer import ORTTrainer, IODescription, ModelDescription
+from onnxruntime.capi.ort_trainer import LossScaler
+import torch
+
+def setup_onnxruntime_with_mpi(args):
+    from mpi4py import MPI
+    comm = MPI.COMM_WORLD
+    args.local_rank = comm.Get_rank()
+    args.world_rank = comm.Get_rank()
+    args.world_size=comm.Get_size()
+    torch.cuda.set_device(args.local_rank)
```

Comment:
it's a bug here. Get_rank gives you the world rank. we need to do set_device(args.local_rank % num_gpus_locally)

Reply:
Updated with args.local_rank = comm.Get_rank() % torch.cuda.device_count()